File: DSA/DSA06030 SẮP XẾP NỔI BỌT - LIỆT KÊ NGƯỢC.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int n;
    cin >> n;
    vector<int> A(n);
    for (auto &&i : A) cin >> i;
    vector<vector<int>> res;
    
    for (int i = 0; i < n; i++)
    {
        bool check = false;
        for (int j = 0; j < n-1; j++)
        {
            if (A[j] > A[j+1])
            {
                swap(A[j], A[j+1]);
                check = true;
            }
        }
        
        if (!check)
        {
            break;
        }

        res.push_back(A);
    }
    
    for (int i = res.size()-1; i >= 0; i--)
    {
        
        cout << "Buoc " << i+1 << ": ";
        for (auto &&k : res[i]) cout << k << " ";
        cout << endl;
    }
    
    
    cout << endl;
}


int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);
    
    int so_bo_test;
    cin >> so_bo_test;
    cin.ignore();
    
    while (so_bo_test--)
    {
        submit();
    }

    return 0;
}

File: DSA_python/DSA06030 SẮP XẾP NỔI BỌT - LIỆT KÊ NGƯỢC.py
def submit():
    n = int(input())
    A = list(map(int, input().split()))
    res = []
    
    for i in range(n):
        check = False
        for j in range(n-1):
            if A[j] > A[j+1]:
                A[j], A[j+1] = A[j+1], A[j]
                check = True
        if check == False:
            break

        res.append(A[:])
    
    for i in range(len(res)-1, -1, -1):
        print(f"Buoc {i+1}:", " ".join(map(str, res[i])))

test = int(input())
for i in range(test):
    submit()
